datalake/src/main/java/com/sequenceiq/datalake/configuration/CDPConfigService.java

```diff
@@ -157,12 +185,14 @@ public String getDefaultRuntime() {
     public List<AdvertisedRuntime> getAdvertisedRuntimes(String cloudPlatform) {
         List<String> runtimeVersions = getDatalakeVersions(cloudPlatform).stream()
                 .filter(runtimeVersion -> advertisedRuntimes.isEmpty() || advertisedRuntimes.contains(runtimeVersion)).collect(Collectors.toList());
+        Optional<String> calculatedDefault
+                = Strings.isNullOrEmpty(this.defaultRuntime) ? runtimeVersions.stream().findFirst() : Optional.ofNullable(this.defaultRuntime);
 
         List<AdvertisedRuntime> advertisedRuntimes = new ArrayList<>();
         for (String runtimeVersion : runtimeVersions) {
             AdvertisedRuntime advertisedRuntime = new AdvertisedRuntime();
             advertisedRuntime.setRuntimeVersion(runtimeVersion);
-            if (runtimeVersion.equals(defaultRuntime)) {
+            if (calculatedDefault.stream().anyMatch(runtimeVersion::equals)) {
```

Comment:
What do you think about `calculatedDefault.map(r -> r.equals(runtimeVersion)).orElse(false)` and no chance that you get an NPE with that?

Reply:
Sure, I can change it but where is the chance to get an NPE?

Reply:
Ahh, ok, you mean runtimeVersion can be null... It can't but we can be super safe :)